### PR TITLE
[windows][dxva] log additional information for dxva troubleshooting, compatibility with WARP/Hyper-V/Graphics Debugger

### PR DESCRIPTION
**Pull request by CrystalP**
## Description

**Commit 1**
Enhanced logging information for dxva rendering without any change in functionality so it could be backported

- Created functions to convert texture format and color space constants to text
New file DirectXHelper.cpp created for that, as that's not win32 stuff for WIN32Util and the functions seemed heavy to inline like the rest of DirectXHelper.h. dxerr.h/.cpp are almost straight copies from MS
- Added enumerations of the color spaces supported by the swap chain
- Added enumeration of the input/output texture formats supported by the dxva processor
- Activated debugging information layer of the separate D3D11 device created for the dxva decoder (mostly c/p from main device created in DeviceResources.cpp)
- In the rendering code paths:
  - Changed logging messages to display the texture format as text instead of number
  - Log more operations that return HRESULT
  - Attempt to translate the HRESULT to text in the log

**Commit 2**
Changed arguments of the main D3D11Device creation: Neither Visual Studio's Graphics Debugger nor the WARP device support video features and they don't accept D3D11_CREATE_DEVICE_VIDEO_SUPPORT.
The Graphics Debugger uses a special capture adapter with vendor Microsoft and I added the Vendor ID to the known list.

## Motivation and context

Help with the troubleshooting of odd problems reported by users regarding dxva
While testing I noticed the issues with the Graphics Debugger, which comes very handy to debug gui rendering problems, and the warp device fallback that hasn't worked for a while.

## How has this been tested?

Tested on Windows 10.

Launch Kodi in Graphics Debugger -> works with with hardware device and doesn't fall back on the WARP device.
Forced a situation to fall back on the WARP device, which also works. 

Read the debug log afterwards, texture formats and color spaces were in text as expected, the enumerations of supported texture formats and color spaces were as expected, there were messages from the d3d debug layer.
I could not make everything fail just to check each and every log message...

## What is the effect on users?

This makes dxva problems hopefully a little easier to troubleshoot when unable to reproduce the issue of the user on own computer.

Kodi can be used in Visual Studio's Graphics Debugger and should fallback on the warp device when the normal device fails to initialize, instead of just exiting.

As a nice side-effect it enables Kodi to run in Hyper-V environments.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp [hunk 1]
```diff
@@ -540,12 +575,14 @@ bool CContext::CreateSurfaces(const D3D11_VIDEO_DECODER_DESC& format, uint32_t c
     texDesc.MiscFlags |= D3D11_RESOURCE_MISC_SHARED;
   }
 
-  CLog::Log(LOGDEBUG, "DXVA: allocating {} surfaces with format {}.", count, format.OutputFormat);
+  CLog::Log(LOGDEBUG, "DXVA: allocating {} surfaces with format {} {}.", count, format.OutputFormat,
+            DX::DXGIFormatToString(format.OutputFormat));
```
Comment (thexai) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp:579]: Maybe better drop format number from log (string name only)

File: xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp [hunk 2]
```diff
@@ -346,7 +348,7 @@ bool CRendererBase::CreateIntermediateTarget(unsigned width, unsigned height, bo
   if (m_IntermediateTarget.Get())
     m_IntermediateTarget.Release();
 
-  CLog::LogF(LOGDEBUG, "intermediate target format {}.", format);
+  CLog::LogF(LOGDEBUG, "intermediate target format {} {}.", format, DX::DXGIFormatToString(format));
```
Comment (thexai) [inline xbmc/cores/VideoPlayer/VideoRenderers/windows/RendererBase.cpp:351]: Same here: number and string seems redundant

**Review by thexai — COMMENTED** [review]
Good improvement, with this the logs provide more information and are more easily understood.

**Comment by thexai** [comment]
Here format can also converted to string:
https://github.com/xbmc/xbmc/blob/aaf4ffbcccc2dde2b39684c0206c8056aeca46b8/xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp#L1170

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp (hunk 1 above)
Comment (CrystalP) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp:579]: yes I was thinking about debugging in VS but after checking there is no need.
I'll do the other ones as well.

**Comment by thexai** [comment]
Another place for use string:
https://github.com/xbmc/xbmc/blob/2d02b97308986fa12f1dafad3f8dfaee05fc2973/xbmc/cores/VideoPlayer/DVDCodecs/Video/DXVA.cpp#L478-L479

**Comment by CrystalP** [comment]
ok I think that's all for the dxva code with the two you pointed out

Jenkins suggestion makes no sense. It doesn't follow the alignment of the file and making it behave "normally" requires reformatting all of DirectXHelper.h so I'd rather leave things like that.

File: xbmc/rendering/dx/DirectXHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    CASEDXGIFORMAT(NV11)
+    CASEDXGIFORMAT(AI44)
+    CASEDXGIFORMAT(IA44)
+    CASEDXGIFORMAT(P8)
+    CASEDXGIFORMAT(A8P8)
+    CASEDXGIFORMAT(B4G4R4A4_UNORM)
+    CASEDXGIFORMAT(P208)
+    CASEDXGIFORMAT(V208)
+    CASEDXGIFORMAT(V408)
+    CASEDXGIFORMAT(SAMPLER_FEEDBACK_MIN_MIP_OPAQUE)
```
Comment (thexai) [inline xbmc/rendering/dx/DirectXHelper.cpp:140]: UWP uses SDK 10.0.18362.0 which seems not support `SAMPLER_FEEDBACK_MIN_MIP_OPAQUE` and `SAMPLER_FEEDBACK_MIP_REGION_USED_OPAQUE` 

Should be removed or added inside `#ifdef TARGET_WINDOWS_DESKTOP` to able compile UWP

Comment (CrystalP) [reply xbmc/rendering/dx/DirectXHelper.cpp]: Yes I'll remove them. They're for D3D12 features it seems.

**Comment by thexai** [comment]
Tested on a Hyper-V VM where before it didn't work due to limitations in video acceleration and now it works fine with "Microsoft Basic Render Video". Even plays 1080p video very well using SW decoding.

VM

**Review by thexai — APPROVED** [review]

**Pull request merged**